index.js

```diff
@@ -2613,6 +2613,42 @@
   }
   S.ifElse = def('ifElse', {}, [Pred(a), Fn(a, b), Fn(a, b), a, b], ifElse);
 
+  //# when :: (a -> Boolean) -> (a -> b) -> a -> b
```

Comment:
I suspect `(a -> Boolean) -> (a -> a) -> a -> a` will not prove very useful (and that the Ramda version is probably often used not according to its type signature) but am happy to give it a go and see if I'm proved wrong